tests/std/tests/VSO_0180469_ptr_cat/test.cpp

```diff
@@ -212,61 +218,82 @@ void ptr_cat_test_cases() {
     test_ptr_cat<0, bool_enum, char>();
 }
 
-template <bool Expected, class Ptr1, class Ptr2, class Pr>
-void test_case_Equal_memcmp_is_safe_helper() {
-    STATIC_ASSERT(_Equal_memcmp_is_safe<Ptr1, Ptr2, Pr> == Expected);
-}
-
-template <bool Expected, class Elem1, class Elem2>
-void test_case_Equal_memcmp_is_safe() {
+template <bool Expected, class Elem1, class Elem2, class Pr>
+void test_case_Equal_memcmp_is_safe_comparator() {
     // Default case
-    test_case_Equal_memcmp_is_safe_helper<Expected, Elem1*, Elem2*, equal_to<>>();
+    STATIC_ASSERT(_Equal_memcmp_is_safe<Elem1*, Elem2*, Pr> == Expected);
     // Adding const should not change the answer
-    test_case_Equal_memcmp_is_safe_helper<Expected, const Elem1*, Elem2*, equal_to<>>();
-    test_case_Equal_memcmp_is_safe_helper<Expected, Elem1*, const Elem2*, equal_to<>>();
-    test_case_Equal_memcmp_is_safe_helper<Expected, const Elem1*, const Elem2*, equal_to<>>();
+    STATIC_ASSERT(_Equal_memcmp_is_safe<const Elem1*, Elem2*, Pr> == Expected);
+    STATIC_ASSERT(_Equal_memcmp_is_safe<Elem1*, const Elem2*, Pr> == Expected);
+    STATIC_ASSERT(_Equal_memcmp_is_safe<const Elem1*, const Elem2*, Pr> == Expected);
+    // Top level const should not change the answer
+    STATIC_ASSERT(_Equal_memcmp_is_safe<Elem1* const, Elem2*, Pr> == Expected);
+    STATIC_ASSERT(_Equal_memcmp_is_safe<Elem1*, Elem2* const, Pr> == Expected);
+    STATIC_ASSERT(_Equal_memcmp_is_safe<Elem1* const, Elem2* const, Pr> == Expected);
+    STATIC_ASSERT(_Equal_memcmp_is_safe<Elem1* const, const Elem2*, Pr> == Expected);
+    STATIC_ASSERT(_Equal_memcmp_is_safe<Elem1*, const Elem2* const, Pr> == Expected);
+    STATIC_ASSERT(_Equal_memcmp_is_safe<Elem1* const, const Elem2* const, Pr> == Expected);
+    STATIC_ASSERT(_Equal_memcmp_is_safe<const Elem1* const, Elem2*, Pr> == Expected);
+    STATIC_ASSERT(_Equal_memcmp_is_safe<const Elem1*, Elem2* const, Pr> == Expected);
+    STATIC_ASSERT(_Equal_memcmp_is_safe<const Elem1* const, Elem2* const, Pr> == Expected);
+    STATIC_ASSERT(_Equal_memcmp_is_safe<const Elem1* const, const Elem2* const, Pr> == Expected);
+    // Iterators that are essentially pointers should not change the answer

```

Comment:
Does "Iterators that are essentially pointers" mean "contiguous iterators", or something else? (Also on 250)

Reply:
:+1: